packages/head-tags/src/utils/index.ts

```diff
@@ -0,0 +1,153 @@
+import { URL } from "frontity";
+import {
+  HeadTags,
+  HeadTag,
+  State,
+  PostTypeWithHeadTags,
+  TaxonomyWithHeadTags
+} from "../../types";
+
+// Attributes that could contain links.
+const possibleLink = ["href", "content"];
+
+const deepTransform = (obj: object, func: Function, ...args: object[]) => {
+  for (const key in obj) {
+    if (typeof obj[key] === "string") {
+      obj[key] = func(obj[key], ...args);
+    } else if (typeof obj[key] === "object") {
+      deepTransform(obj[key], func, ...args);
+    }
+  }
+};
+
+const getWpUrl = (api: string, isWpCom: boolean): URL => {
+  const apiUrl = new URL(api);
+  if (isWpCom) {
+    const { pathname } = apiUrl;
+    return new URL(pathname.replace(/^\/wp\/v2\/sites\//, "https://"));
+  }
+  // Get API subdirectory.
+  const apiSubdir = apiUrl.pathname.replace(/\/wp-json\/?$/, "/");
```

Comment:
I think we misunderstood `state.source.subdirectory` the other day, because it's the Frontity site's subdirectory and not the WordPress site's, so we would have to add a new setting for that.